src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -251,4 +253,14 @@ private void validateOperationProperties(@Nullable Schema schema) {
         "Fields %s are in the table key, but not in the input schema.", result), NAME_TABLE_KEY);
     }
   }
+
+  private void validateWriteDisposition() {
+    if (Arrays.stream(WriteDisposition.values()).map(Enum::name).noneMatch(writeDisposition.toUpperCase()::equals)) {
```

Comment:
after the change to boolean, this won't need to happen. But we should add some validation to make sure truncate is false when the operation is 'update'.

Reply:
I moved the validation to the abstract class since it applies to both BigQuery sinks.